src/state/swap.ts

```diff
@@ -23,6 +23,31 @@ export const swapAtom = atomWithImmer<Swap>({
   [Field.INPUT]: nativeOnChain(SupportedChainId.MAINNET),
 })
 
+export interface IsControlledSwapState {
+  isControlledAmount?: boolean
+  isControlledToken?: boolean
+}
+
+export const isControlledSwapStateAtom = atom<IsControlledSwapState>({
+  isControlledAmount: false,
+  isControlledToken: false,
+})
```

Comment:
is possible for integrator to use controlled token state & uncontrolled amount state; need to disambiguate between the two